### PR TITLE
boards: arm: gigadevice: Initial support for gd32f470ik

### PR DESCRIPTION
**Pull request by duchesne9al**
Add initial support for the GD32F470I evaluation board.

There is an open [pull request](https://github.com/zephyrproject-rtos/hal_gigadevice/pull/23) to add pinctrl headers for GD32F470xx devices required for: 
`#include <dt-bindings/pinctrl/gd32f470i(g-i-k)xx-pinctrl.h>`  in _gd32f470i_eval_defconfig_ file.

I put the compatible device **GD32F450IK** in the board.cmake file, **GD32F470IK** is not yet supported by the SEGGER JLink firmware.

**Samples tested on my board:** 
sample/hellow_world  
sample/samples/basic/blinky  

Signed-off-by: Alexandre Duchesne [alexandre.duchesne@rtone.fr](mailto:alexandre.duchesne@rtone.fr)

### CODE REVIEW

File: boards/arm/gd32f470i_eval/board.cmake [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Copyright (c) 2022, Rtone.
+# SPDX-License-Identifier: Apache-2.0
+
+board_runner_args(jlink "--device=GD32F450IK" "--speed=4000")
```
Comment (gmarull) [inline boards/arm/gd32f470i_eval/board.cmake:4]: this looks wrong

**Review by gmarull — CHANGES_REQUESTED** [review]
lgtm, one minor comment. please update hal sha (or `pull/23/head`) in west.yaml as well.

File: boards/arm/gd32f470i_eval/board.cmake (hunk 1 above)
Comment (gmarull) [reply boards/arm/gd32f470i_eval/board.cmake:4]: I've seen the comment: please add a comment mentioning segger doesn't support f470 yet but that this works (?)

**Review by nandojve — CHANGES_REQUESTED** [review]
Hi @duchesne9al,

I have a few remarks and would like know if you can provide a better photo with better framing?

File: boards/arm/gd32f470i_eval/board.cmake [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# Copyright (c) 2022, Rtone.
+# SPDX-License-Identifier: Apache-2.0
+
+# GD32F470xx series is not yet supported by SEGGER J-Link
+board_runner_args(jlink "--device=GD32F450IK" "--speed=4000")
```
[the remark this answers is not in the document]
Comment (duchesne9al) [reply boards/arm/gd32f470i_eval/board.cmake:5]: Yes

**Comment by duchesne9al** [comment]
Hi @nandojve,

After building **gd32f470i_eval** board I get a `.config` file with  `CONFIG_SOC="gd32f450"`.  
It looks like  `Kconfig.defconfig.gd32f470` is being ignored.
Do you see anything wrongly declared or a missing step when adding a SoC?
Thanks.

**Comment by duchesne9al** [comment]
> Hi @duchesne9al,
> 
> I have a few remarks and would like know if you can provide a better photo with better framing?

Hi @nandojve,

Yes the framing is not great, I tried to avoid my reflection on the LCD screen but I'll see what I can do tomorrow with the light of day :)

File: dts/arm/gigadevice/gd32f4xx/gd32f470.dtsi [hunk 3]
```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2022, Rtone.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <gigadevice/gd32f4xx/gd32f4xx.dtsi>
+
+/ {
+
```
Comment (gmarull) [inline dts/arm/gigadevice/gd32f4xx/gd32f470.dtsi:10]: note: these changes are part of the board commit, but they are not strictly related (like SoC changes), please split into its own commit

Comment (duchesne9al) [reply dts/arm/gigadevice/gd32f4xx/gd32f470.dtsi:10]: done.

File: dts/arm/gigadevice/gd32f4xx/gd32f470.dtsi [hunk 4]
```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2022, Rtone.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <gigadevice/gd32f4xx/gd32f4xx.dtsi>
```
Comment (nandojve) [inline dts/arm/gigadevice/gd32f4xx/gd32f470.dtsi:7]: Since F470 is F450 with more speed and SRAM, you can just include [gd32f450.dtsi](https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/arm/gigadevice/gd32f4xx/gd32f450.dtsi) here.

**Review by nandojve — CHANGES_REQUESTED** [review]

File: dts/arm/gigadevice/gd32f4xx/gd32f470.dtsi (hunk 4 above)
Comment (duchesne9al) [reply dts/arm/gigadevice/gd32f4xx/gd32f470.dtsi]: done.

**Review by gmarull — APPROVED** [review]

**Review by nandojve — APPROVED** [review]
